src/blueapi/service/interface.py

```diff
@@ -20,52 +18,81 @@
 context and worker"""
 
 
-class _Singleton:
-    context: BlueskyContext
-    worker: Worker
-    messaging_template: MessagingTemplate | None = None
-    initialized = False
+@runtime_checkable
+class CacheClearable(Protocol):
```

Comment:
I'm not sure how this is used but can leave it in for now.

Reply:
It is literally for duck typing, I am at once asserting that these things have a `cache_clear` method and using it. It doesn't add much safety, but hopefully a bit of readability.